### PR TITLE
New Backend & Location Handling

### PR DESCRIPTION
**Pull request by thisjustin123**
## Overview:

This PR is primarily focused on connecting Uplift to the new backend API but also basically includes everything we need for full functionality at launch. 

## Changes:

- `ApolloAdapters` have been restructured to work with the new backend. This includes all new GraphQL schema stuff.
- Teagle Up and Teagle Down now appear as separate gyms with their own hours as intended.
- Location & Permissions flow has been added. When location is granted, location data automatically populates through to the rest of the app. (See demo below; this is what happens when you use a data flow architecture! Side eye @ eatery..)

## Demo:
<details>
  <summary>Location Demo</summary>

I showcase giving permissions for the first time, as well as denying permissions.

Note how after I press "Allow" for the permissions, Teagle and Helen automatically swap since Teagle is closer. Also, all distance information automatically pops up on the home cards:

https://github.com/cuappdev/uplift-android/assets/47724806/2358b54a-62aa-4fb2-9dc5-193045c8da7e

</details>

<details>
  <summary>Pull to Refresh</summary>

https://github.com/cuappdev/uplift-android/assets/47724806/b128f4d3-ebf7-46a0-a5ab-f51bac25182d

</details>

### CODE REVIEW

File: app/src/main/java/com/cornellappdev/uplift/networking/ApolloAdapters.kt [hunk 1]
```diff
@@ -1,265 +1,182 @@
...
+            id = facility.id,
+            popularTimes = pullPopularTimes(facility),
+            // Need replace because there's a typo with the single quote.
+            imageUrl = imageUrl?.replace("'", "") ?: defaultGymUrl,
+            hours = pullHours(facility),
+            equipmentGroupings = pullEquipmentGroupings(facility),
+            miscellaneous = pullMiscellaneous(),
+            bowlingInfo = pullBowling(),
+            // No swimming info in backend yet...
+            swimmingInfo = pullSwimmingInfo(),
```
Comment (emilyhu8) [inline app/src/main/java/com/cornellappdev/uplift/networking/ApolloAdapters.kt:169]: do you know when swimming will be added lol

File: app/src/main/java/com/cornellappdev/uplift/networking/UpliftApiRepository.kt [hunk 2]
```diff
@@ -67,15 +66,16 @@ object UpliftApiRepository {
...
                     val gymList = it.data?.gyms?.filterNotNull()
                     if (gymList == null) {
                         ApiResponse.Error
                     } else {
-                        ApiResponse.Success(gymList.map { query -> query.toUpliftGym() })
+                        ApiResponse.Success(gymList.map { query -> query.toUpliftGyms() }.flatten())
                     }
                 }
                 .catch {
+                    it.printStackTrace()
```
Comment (emilyhu8) [inline app/src/main/java/com/cornellappdev/uplift/networking/UpliftApiRepository.kt]: do we need this printStackTrace? I feel like its better if we delete

File: app/src/main/java/com/cornellappdev/uplift/ui/components/general/NoClasses.kt [hunk 3]
```diff
@@ -31,22 +33,23 @@ fun NoClasses() {
...
-            fontWeight = FontWeight(400),
-            fontSize = 17.sp,
-            lineHeight = 17.07.sp,
-            textAlign = TextAlign.Center,
-            color = PRIMARY_BLACK,
-            fontFamily = montserratFamily
-        )
+        if (!comingSoon)
+            Text(
+                text = "Relax with some tea or play a sport",
```
Comment (emilyhu8) [inline app/src/main/java/com/cornellappdev/uplift/ui/components/general/NoClasses.kt:46]: aha this is some funny text

Comment (emilyhu8) [reply app/src/main/java/com/cornellappdev/uplift/ui/components/general/NoClasses.kt:46]: u ate

File: app/src/main/java/com/cornellappdev/uplift/ui/components/gymdetail/GymTodaysClasses.kt [hunk 4]
```diff
@@ -59,7 +63,8 @@ fun GymTodaysClasses(
         }
         if (todaysClasses.value.isEmpty()) {
             Text(
-                text = "We are done for today.\nPlease check again tomorrow!",
+                text = "Classes are coming soon!\nStay tuned for an Uplift update.",
+                // TODO: Change to: text = "We are done for today.\nPlease check again tomorrow!",
```
Comment (emilyhu8) [inline app/src/main/java/com/cornellappdev/uplift/ui/components/gymdetail/GymTodaysClasses.kt:67]: can you not change the text right now?

File: app/src/main/java/com/cornellappdev/uplift/ui/viewmodels/GymDetailViewModel.kt [hunk 5]
```diff
@@ -70,5 +71,7 @@ class GymDetailViewModel : ViewModel() {
         stack.pop()
         if (stack.isNotEmpty())
             _gymFlow.value = stack.peek()
+        else
+            Log.i("stack", "Empty :-)")
```
Comment (emilyhu8) [inline app/src/main/java/com/cornellappdev/uplift/ui/viewmodels/GymDetailViewModel.kt]: delete log

**Review by emilyhu8 — APPROVED** [review]
great work! just some comments about logs lol

File: app/src/main/java/com/cornellappdev/uplift/ui/components/gymdetail/GymTodaysClasses.kt (hunk 4 above)
Comment (thisjustin123) [reply app/src/main/java/com/cornellappdev/uplift/ui/components/gymdetail/GymTodaysClasses.kt:67]: Not yet, cuz classes aren't in the backend yet. If I put it as text that indicates that there are no more classes today, that's a bit misleading. I'll change it back when the classes are added to backend.

File: app/src/main/java/com/cornellappdev/uplift/networking/ApolloAdapters.kt (hunk 1 above)
Comment (thisjustin123) [reply app/src/main/java/com/cornellappdev/uplift/networking/ApolloAdapters.kt:169]: Oh that was an old comment when we had the old backend. In fact a LOT of stuff isn't in backend yet lol hence the like million placeholder functions. I don't think swimming info's coming for a while.

File: app/src/main/java/com/cornellappdev/uplift/ui/components/general/NoClasses.kt (hunk 3 above)
Comment (thisjustin123) [reply app/src/main/java/com/cornellappdev/uplift/ui/components/general/NoClasses.kt:46]: lol tell that to design, i just typed it in for them

**Pull request merged**